File: scripts/snakes_ladders.py
import random
nPlayers = 4
endSquare = 100
snakes = { 14:3, 28:7, 34:12, 68:42, 85:23, 96:67 }
ladders = { 5:16 , 22:56 , 38:66 ,51:76 ,64:91 ,72:84 }

# create vector of current positions
P = [1]*4

# specify starting player
i = 0

while max(P) < endSquare:

    D = random.randint(1,6)
    print(D)
    P[i] = P[i]+D
    if P[i] in snakes:
        P[i] = snakes[P[i]]
        print("Player " + str(i + 1)  + " goes down a snake.") 
    elif P[i] in ladders:
        P[i] = ladders[P[i]]
        print("Player " + str(i + 1)  + " goes up a ladder.") 
    
    print("Player " + str(i + 1)  + " is at square " + str(P[i])) 
 
    if i == (nPlayers - 1):
        i = 0
    else:
        i+=1

File: scripts/snakes_ladders.R
nPlayers <- 4
endSquare <- 100
snakes<-data.frame("Top" = c(14,28,34,68,85,96), "Bottom" = c(3,7,12,42,23,67))
ladders<-data.frame("Bottom" = c(5,22,38,51,64,72), "Top" = c(16,56,66,76,91,84))

# create vector of current positions
P<-rep(1, nPlayers)

# specify starting player
i<-1

while(max(P) < endSquare){

    D <- sample(c(1:6),1)
    P[i]<-P[i]+D
    if( P[i] %in% snakes$Top){
        P[i]<-snakes$Bottom[snakes$Top== P[i]]
    } else if (P[i] %in% ladders$Bottom) {
        P[i]<-ladders$Top[ladders$Bottom == P[i]]
    }

    print(paste("Player", i, "is at square", P[i])) 
 
   if(i == nPlayers){
        i <- 1
    } else {
        i <- i+1
    }
}